src/utils/reader_chunks.ts

```diff
@@ -176,13 +176,15 @@ class BufferChunksReader {
 
   readVarLengthBuffer() {
     const len = this.readVarintNum();
-    const buf = this.read(len);
-    if (buf.length !== len) {
-      throw new Error(
-        `Invalid length while reading varlength buffer. Expected to read: ${len} and read ${buf.length}`
-      );
+    if (len) {
+      const buf = this.read(len);
+      if (buf?.length !== len) {
+        throw new Error(
+          `Invalid length while reading varlength buffer. Expected to read: ${len} and read ${buf?.length}`
+        );
+      }
+      return buf;
     }
-    return buf;
   }
 
   readVarintBuf() {
```

Comment:
`this.buf` does not exist. @kevinejohn can you help here?

Reply:
it just just be a local `const buf` and not `this.buf`. Am I missing something?